packages/core/src/auth/Logout.ts

```diff
@@ -25,24 +25,52 @@ export class Logout {
      * @static
      * @param {AbstractSession} session
      * @returns
-     * @memberof Login
+     * @memberof Logout
      */
     public static async apimlLogout(session: AbstractSession) {
         Logger.getAppLogger().trace("Logout.logout()");
         ImperativeExpect.toNotBeNullOrUndefined(session, "Required session must be defined");
-        ImperativeExpect.toBeEqual(session.ISession.tokenType, "apimlAuthenticationToken",
-            "Token type for API ML logout must be apimlAuthenticationToken.");
+        ImperativeExpect.toMatchRegExp(session.ISession.tokenType, "^apimlAuthenticationToken.*",
+            `Token type (${session.ISession.tokenType}) for API ML logout must start with 'apimlAuthenticationToken'.`);
         ImperativeExpect.toNotBeNullOrUndefined(session.ISession.tokenValue, "Session token not populated. Unable to log out.");
 
         const client = new ZosmfRestClient(session);
+        if(!session.ISession.basePath) {
+            session.ISession.basePath = "/"; // prevent basepath requirement error on invalid credentials
+        }
         try{
             await client.request({
                 request: "POST",
                 resource: LogoutConstants.APIML_V1_RESOURCE
             });
         } catch (err) {
-            if (!err.message.includes(LogoutConstants.APIML_V1_TOKEN_EXP_ERR)) {
-                throw err;
+            let errorToThrow = err;
+            for (const errorKey in LogoutConstants.APIML_V2_LOGOUT_ERR_LIST) {
+                if (err.message.includes(LogoutConstants.APIML_V2_LOGOUT_ERR_LIST[errorKey])) {
+                    switch (errorKey) {
+                        case "V2_TOKEN_INVALID": // Token is invalid (logged out)
+                        case "V2_TOKEN_EXPIRED": // Token expired (trully expired)
+                        case "V2_TOKEN_MISSING": // Token type is not ^apimlAuthenticationToken.*
+                            errorToThrow = new ImperativeError({
+                                msg: "Token is not valid or expired.\n" +
+                                    "For CLI usage, see `zowe auth logout apiml --help`",
+                                errorCode: client.response.statusCode.toString()
+                            });
+                            break;
+                        case "V1_TOKEN_EXPIRED":
+                        default:
+                            errorToThrow = null;
+                            break;
+                    }
+                }
+            }
+
+            if (errorToThrow) {
+                throw errorToThrow;
+            }
```

Comment:
Where possible I think we should prefer checking HTTP error codes (well-defined behavior) over looking for specific substrings in the error message (subject to change).

I think we discussed IRL that it may be possible to replace the "V2_TOKEN" checks with a single check for error code 401 🙂 But maybe I am confused or remembering wrong?

Reply:
added a note for clarity (and the block of code that we could replace all of the `V2_TOKEN_...` checks with) 😋